inst/include/Rcpp/macros/macros.h

```diff
@@ -98,7 +98,10 @@
   catch (...) {                                                                \
     return string_to_try_error("c++ exception (unknown reason)");              \
   }                                                                            \
-  return R_NilValue;
+  UNPROTECT(nprot);                                                            \
+  return R_NilValue;                                                           \
+  if (stop_sym == R_NilValue) /* never reached but */                          \
```

Comment:
As a side note: is `(void) stop_sym;` also sufficient to silence the warning? Just because IIUC that's the gcc-approved way to "use" a variable to silence this particular diagnostic.

Just wonder if in theory a static analyzer would complain about 'dead code' here. Or in theory a newer version of gcc would detect that.

Anyway. This is mostly just bike-shedding so feel free to ignore :-)

Reply:
I like that.  I had not seen that trick (or if i had, have forgotten about it).  Let me give it a quick spin.